### DISCUSSION TITLE
Problems with Markdown syntax highlighting

### DISCUSSION THREAD

**Discussion by MaralWa** [epwalsh/obsidian.nvim#394] [Q&A] [answered]
Hi,

I am a long time VimWiki user but considering to replace it with obsidian.nvim.

I removed VimWiki and all corresponding settings from my LayzVim configuration and added obsidian.nvim. My config is still very basic:

```lua
  {
    "epwalsh/obsidian.nvim",
    version = "*",  -- recommended, use latest release instead of latest commit
    lazy = true,
    ft = "markdown",
    dependencies = {
      -- Required.
      "nvim-lua/plenary.nvim",
    },
    opts = {
      workspaces = {
        { name = "Wiki", path = '~/Documents/vimwiki/' },
      },
    },
```

But markdown header lines are not highlighted correctly. Here's an example:

<img width="1817" alt="nvim_obsidian" src="https://github.com/epwalsh/obsidian.nvim/assets/18118847/0bccbcd0-0109-4286-a4f9-bb9e28d2ee3a">

On the left side nvim shows a note that I created with the command `ObsidianNew`. The header lines `# Test`, `## Section`, `### Subsection` and `# Obsidian` are not highlighted but instead displayed as if the cursor was placed on those lines. Below each of these lines a line of question marks is shown although these lines are empty. Also the first task is not rendered correctly. On th right the same note is displayed by Obsidian.

The following screenshot shows the same note. But in this case nvim used my old VimWiki configuration:

<img width="1817" alt="nvim_wiki" src="https://github.com/epwalsh/obsidian.nvim/assets/18118847/4009de73-77d5-415f-90b7-f00d852ad8b6">

Here the header lines were highlighted correctly. 

In both cases I run nvim inside `Oh My Zsh` running in an `iterm2` window. The color scheme is GruvBox Dark. iterm2 uses the font `MesloLGL Nerd Font Mono`.

Ciao

Maral

**Comment by epwalsh** [comment, marked as answer]
Hey @MaralWa, do you use nvim-treesitter? I recently had this issue after upgrading treesitter and found that there were some breaking changes that my theme plugin hadn't updated for yet. Pinning treesitter to the last release fixed it for me.

**Reply by MaralWa** [reply]
Thanks for the hint. I think it was indeed caused by some treesitter commit. When I switched back to my VimWiki configuration, the issue also occured. I could not identify the commit or version that caused the issue. So I gave up and was quite frustrated. But after some days the issue resolved itself magically by some other plugin updates. Hence i did not inverst more time.
